app/models/name.rb

```diff
@@ -2555,29 +2561,75 @@ def save_with_log(log = nil, args = {})
     save
   end
 
-  ################################################################################
-
   protected
 
+  ##############################################################################
+  #
+  #  :section: Limits
+  #
+  ##############################################################################
+
+  # :string field size limits in characters, based on this algorithm:
+  # Theoretical max differences between (text_name + author) and the field
+  #   search_name: 4
+  #     Adding a " sp." at the end is the worst case.
+  #   sort_name:   21
+  #     It adds " {N" and " !" (5 chars) in front of subgenus and the epithet
+  #     that goes with it. There can be up to 4 infrageneric ranks (subgenus,
+  #     section, subsection, stirps, plus the space between name and author.
+  #     That adds up to 5*4 + 1 = 21.
+  #   display_name: 41
+  #     Adds the space between name and author; "**__" and "__**" (8 chars)
+  #     around every epithet, grouping genus and species. Infrageneric ranks
+  #     win, making as many as 5 separate bold epithets or epithet pairs.
+  #     That adds up to 8*5 + 1 = 41.
+
+  # Numbers are hard-coded (rather than calculated) to make it easier to copy
+  # them to migrations.
+  def self.author_limit
+    # max if text_name_limit == 100, and we want to keep sort_name_limit < 256
+    134
+  end
+
+  def self.search_name_limit
+    238
+  end
+
+  def self.sort_name_limit
+    255
+  end
+
+  def self.display_name_limit
+    285
+  end
+
+  def self.text_name_limit
+    100
+  end
```

Comment:
It would be nice to grab these numbers from the db (instead of hard-coding them).  But I couldn't figure out how to do that.